extension/data.ts

```diff
@@ -277,112 +295,80 @@ class RcxDict {
               }
               o = { word: word, type: 0xff, reason: '' } as Deinflection;
               if (have[newWord] !== undefined) {
-                o = r[have[newWord]];
+                o = possibleDeinflections[have[newWord]];
                 o.type |= rule.typeMask >> 8;
 
                 continue;
               }
-              have[newWord] = r.length;
-              if (r[i].reason.length) {
+              have[newWord] = possibleDeinflections.length;
+              if (possibleDeinflections[i].reason.length) {
                 o.reason =
-                  this.difReasons[rule.reasonIndex] + ' &lt; ' + r[i].reason;
+                  this.difReasons[rule.reasonIndex] +
+                  ' &lt; ' +
+                  possibleDeinflections[i].reason;
               } else {
                 o.reason = this.difReasons[rule.reasonIndex];
               }
               o.type = rule.typeMask >> 8;
               o.word = newWord;
-              r.push(o);
+              possibleDeinflections.push(o);
             }
           }
         }
       }
-    } while (++i < r.length);
+    } while (++i < possibleDeinflections.length);
 
-    return r;
+    return possibleDeinflections;
   }
 
-  // katakana -> hiragana conversion tables
-  ch: number[] = [
-    0x3092, 0x3041, 0x3043, 0x3045, 0x3047, 0x3049, 0x3083, 0x3085, 0x3087,
-    0x3063, 0x30fc, 0x3042, 0x3044, 0x3046, 0x3048, 0x304a, 0x304b, 0x304d,
-    0x304f, 0x3051, 0x3053, 0x3055, 0x3057, 0x3059, 0x305b, 0x305d, 0x305f,
-    0x3061, 0x3064, 0x3066, 0x3068, 0x306a, 0x306b, 0x306c, 0x306d, 0x306e,
-    0x306f, 0x3072, 0x3075, 0x3078, 0x307b, 0x307e, 0x307f, 0x3080, 0x3081,
-    0x3082, 0x3084, 0x3086, 0x3088, 0x3089, 0x308a, 0x308b, 0x308c, 0x308d,
-    0x308f, 0x3093,
-  ];
+  isKana(charCode: number): boolean {
+    return (
+      (charCode >= KANA.HIRAGANA_START && charCode <= KANA.HIRAGANA_END) ||
+      (charCode >= KANA.KATAKANA_START && charCode <= KANA.KATAKANA_END) ||
+      (charCode >= KANA.HW_KATAKANA_START && charCode <= KANA.HW_KATAKANA_END)
+    );
+  }
 
-  cv: number[] = [
-    0x30f4, 0xff74, 0xff75, 0x304c, 0x304e, 0x3050, 0x3052, 0x3054, 0x3056,
-    0x3058, 0x305a, 0x305c, 0x305e, 0x3060, 0x3062, 0x3065, 0x3067, 0x3069,
-    0xff85, 0xff86, 0xff87, 0xff88, 0xff89, 0x3070, 0x3073, 0x3076, 0x3079,
-    0x307c,
-  ];
-  cs: number[] = [0x3071, 0x3074, 0x3077, 0x307a, 0x307d];
+  convertToHiragana(str: string): string {
```

Comment:
I might have said this elsewhere but definititely recommend a jsdoc for this method and maybe a tweak to the name since it also has a truncation function now (which we added back for consistency)

Reply:
What do you mean by `truncation` portion? If you mean the stripping of the ZWJ and Tilde, etc... then that happens before it is sent to the `convertToHiragana` method.

Here is what I currently have using TS-Doc:

<img width="826" alt="Screenshot 2023-03-08 at 7 06 05 AM" src="https://user-images.githubusercontent.com/81494248/223749638-07e2ff15-84f0-4c88-b106-024f16e4b415.png">

What do you think?

Reply:
It's at line 352; I left a comment there for easy reference.

The TS doc is solid but I'll leave some pedantic comments (mostly conforming to best practices developed in my day job but that have served me well)